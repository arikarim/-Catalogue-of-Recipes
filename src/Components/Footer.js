import React from 'react';
import { Link } from 'react-router-dom';
import './Component.css';

const Footer = () => (
  <footer className="header text-decoration-none ">
    <Link className="link" to="/"><h1 className="header__title text-decoration-none">FoodRecipe</h1></Link>
    <span className="span">Ari Karim</span>
  </footer>
);

export default Footer;
