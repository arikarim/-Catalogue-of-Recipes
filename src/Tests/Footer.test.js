import React from 'react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Footer from '../Components/Footer';

describe('Footer with Link', () => {
  test('full app rendering/navigating', () => {
    render(<Footer />, { wrapper: MemoryRouter });
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText('FoodRecipe')).toBeInTheDocument;
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Footer />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
