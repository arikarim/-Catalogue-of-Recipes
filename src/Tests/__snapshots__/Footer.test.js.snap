// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer with Link should match with snapshot 1`] = `
<footer
  className="header text-decoration-none"
>
  <a
    className="link"
    href="/"
    onClick={[Function]}
  >
    <h1
      className="header__title text-decoration-none"
    >
      FoodRecipe
    </h1>
  </a>
  <span
    className="span"
  >
    Ari Karim
  </span>
</footer>
`;
